src/main/java/peoplesoft/model/person/Person.java

```diff
@@ -135,6 +135,29 @@ public boolean isSamePerson(Person otherPerson) {
             && getPersonId().equals(otherPerson.getPersonId());
     }
 
+    /**
+     * Returns true if both persons have the data fields.
+     * This defines a weak notion of equality between two persons.
+     */
+    public boolean isDuplicate(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Person)) {
+            return false;
+        }
+
+        Person otherPerson = (Person) other;
+        return otherPerson.getName().equals(getName())
+                && otherPerson.getPhone().equals(getPhone())
+                && otherPerson.getEmail().equals(getEmail())
+                && otherPerson.getAddress().equals(getAddress())
+                && otherPerson.getRate().equals(getRate())
+                && otherPerson.getTags().equals(getTags())
+                && otherPerson.getPayments().equals(getPayments());
```

Comment:
if we don't expect to have to compare `Person`s to *any* kind of `Object`, i think we can have the method take a `Person` instead

(we can't do the same for `equals()` because that's inherited from `Object.equals(Object)`)

```suggestion
    public boolean isDuplicate(Person otherPerson) {
        if (otherPerson == this) {
            return true;
        }
        
        return otherPerson.getName().equals(getName())
                && otherPerson.getPhone().equals(getPhone())
                && otherPerson.getEmail().equals(getEmail())
                && otherPerson.getAddress().equals(getAddress())
                && otherPerson.getRate().equals(getRate())
                && otherPerson.getTags().equals(getTags())
                && otherPerson.getPayments().equals(getPayments());
```